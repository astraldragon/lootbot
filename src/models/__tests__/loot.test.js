const loot = require("../loot");

describe("models/loot", () => {
  beforeEach(() => {
    this.model = {
      belongsTo: jest.fn("belongsTo").mockReturnValue()
    };

    this.sequelize = {
      define: jest.fn("define").mockReturnValue(this.model)
    };

    this.dataTypes = {
      STRING: "string"
    };

    this.models = {
      Tier: jest.fn("Tier")
    };
  });

  it("should create the model definition", () => {
    loot(this.sequelize, this.dataTypes);
    expect(this.sequelize.define).toMatchSnapshot();
  });

  it("should associate loot with tiers", () => {
    const Loot = loot(this.sequelize, this.dataTypes);
    Loot.associate(this.models);

    expect(Loot.belongsTo).toHaveBeenCalledWith(this.models.Tier, {
      foreignKey: "tier_id"
    });
  });
});
