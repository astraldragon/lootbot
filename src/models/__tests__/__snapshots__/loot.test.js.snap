// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`models/loot should create the model definition 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      "Loot",
      Object {
        "guild": "string",
        "name": "string",
      },
      Object {},
    ],
  ],
}
`;
